src/Containers-SmallOrderedSet/CTSmallOrderedSetTest.class.st

```diff
@@ -81,14 +82,242 @@ CTSmallOrderedSetTest >> testDo [
 	self assert: (seen at: 2) equals: 1
 ]
 
-{ #category : #testing }
+{ #category : 'testing' }
+CTSmallOrderedSetTest >> testDoSeparatedBy [ 
+    | seen |
+
+    "Add elements manually, including duplicates"
+    collection add: 2.
+    collection add: 1.
+    collection add: 3.
+    collection add: 2.  "Duplicate"
+    collection add: 1.  "Duplicate"
+
+    "Verify that collection size is correct (duplicates should not be counted)"
+    self assert: collection size equals: 3.
+
+    "Iterate over the collection, inserting separators"
+    seen := Array streamContents: [ :stream |
+        collection do: [ :each | stream nextPut: each ] 
+                  separatedBy: [ stream nextPut: '-' ] ].
+
+    "Ensure separators are added correctly and duplicates are not present"
+    self assert: seen equals: #(2 '-' 1 '-' 3).
+
+]
+
+{ #category : 'testing' }
+CTSmallOrderedSetTest >> testErrorNotFound [ 
+    | collectionSet emptyCollection errorFlagEmpty errorFlagNonExistent errorFlagAlreadyRemoved |
+
+    "Create an instance of CTSmallOrderedSet"
+    collectionSet := CTSmallOrderedSet new: 2.
+
+    "Try removing an element from an empty set"
+    emptyCollection := CTSmallOrderedSet new: 0.
+    errorFlagEmpty := false.
+    ([
+        emptyCollection remove: 42. "Removing from empty set"
+    ]) on: Error do: [ :ex | errorFlagEmpty := true. ].
+    self assert: errorFlagEmpty.  "Ensure an error is raised for empty set"
+
```

Comment:
Can you split the tests into two different ones. 
One for the empty and non empty case.